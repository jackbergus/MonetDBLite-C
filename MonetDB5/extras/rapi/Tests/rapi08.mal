gv:= bat.new(:oid,:int);
bat.append(gv,1);
bat.append(gv,1);
bat.append(gv,2);
bat.append(gv,2);

dv:= bat.new(:oid,:int);
bat.append(dv,42);
bat.append(dv,84);
bat.append(dv,42);
bat.append(dv,21);

# tuple identifiers
ti:= bat.new(:oid,:oid);
bat.append(ti,0:oid);
bat.append(ti,1:oid);
bat.append(ti,2:oid);
bat.append(ti,3:oid);

X_8 := algebra.leftfetchjoin(ti,gv);
(X_9,r1_9,r2_9) := group.subgroupdone(X_8);
X_12 := algebra.leftfetchjoin(r1_9,X_8);
X_15 := algebra.leftfetchjoin(ti,dv); 
X_16:bat[:any,:dbl] := rapi.subeval("res <- aggregate(arg1, by=list(arg2), FUN=median); return(res$x)",X_15,X_9,r1_9,true);

io.print(X_16);

