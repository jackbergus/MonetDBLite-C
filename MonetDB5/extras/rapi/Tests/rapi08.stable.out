stdout of test 'rapi08` in directory 'monetdb5/extras/rapi` itself:


# 12:50:57 >  
# 12:50:57 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=33963" "--set" "mapi_usock=/var/tmp/mtest-14670/.s.monetdb.33963" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/export/scratch1/hannes/fakefs/var/MonetDB/mTests_monetdb5_extras_rapi" "rapi08.mal"
# 12:50:57 >  

# MonetDB 5 server v11.17.2
# This is an unreleased version
# Serving database 'mTests_monetdb5_extras_rapi', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs dynamically linked
# Found 15.592 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2014 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://bristol.ins.cwi.nl:33963/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-14670/.s.monetdb.33963
# MonetDB/GIS module loaded
# MonetDB/JAQL module loaded
# MonetDB/SQL module loaded
function user.main():void;
    gv := bat.new(:oid,:int);
    bat.append(gv,1);
    bat.append(gv,1);
    bat.append(gv,2);
    bat.append(gv,2);
    dv := bat.new(:oid,:int);
    bat.append(dv,42);
    bat.append(dv,84);
    bat.append(dv,42);
    bat.append(dv,21);
# tuple identifiers 
    ti := bat.new(:oid,:oid);
    bat.append(ti,0@0:oid);
    bat.append(ti,1@0:oid);
    bat.append(ti,2@0:oid);
    bat.append(ti,3@0:oid);
    X_8 := algebra.leftfetchjoin(ti,gv);
    (X_9,r1_9,r2_9) := group.subgroupdone(X_8);
    X_12 := algebra.leftfetchjoin(r1_9,X_8);
    X_15 := algebra.leftfetchjoin(ti,dv);
    X_16:bat[:any,:dbl]  := rapi.subeval("res <- aggregate(arg1, by=list(arg2), FUN=median); return(res$x)",X_15,X_9,r1_9,true);
    io.print(X_16);
end main;
#--------------------------#
# h	t  # name
# void	dbl  # type
#--------------------------#
[ 0@0,	63  ]
[ 1@0,	31.5  ]

# 12:50:58 >  
# 12:50:58 >  "Done."
# 12:50:58 >  

